packages/wallets/ledger/src/adapter.ts

```diff
@@ -142,6 +144,22 @@ export class LedgerWalletAdapter extends BaseSignerWalletAdapter {
         }
     }
 
+    async signMessage(message: Uint8Array): Promise<Uint8Array> {
+        try {
+            try {
+                const transport = this._transport;
+                if (!transport) throw new WalletNotConnectedError();
+
+                return new Uint8Array(await signMessage(transport, Buffer.from(message.buffer), this._derivationPath));
```

Comment:
Let's split up the async call and the return to make it easier to understand (and step through).

```suggestion
                const signature = await signMessage(transport, Buffer.from(message.buffer), this._derivationPath);
                return new Uint8Array(signature);
```

Reply:
Refactored.